SWLOR.Game.Server/Feature/ChatCommandDefinition/DMChatCommand.cs

```diff
@@ -406,10 +406,66 @@ private void SetLocalVariable()
 
                     SendMessageToPC(user, "Local string set: " + variableName + " = " + value);
                 });
+
+            _builder.Create("settag")
+                .Description("Sets a local tag on a target.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .AvailableToAllOnTestEnvironment()
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    if (args.Length <= 0)
+                    {
+                        return "Missing arguments. Format should be: /SetTag <VALUE>. Example: /SetTag DUNGEON_ENTRANCE";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    if (!GetIsObjectValid(target))
+                    {
+                        SendMessageToPC(user, "Target is invalid. Please target a creature, placeable or door.");
+                    }
+                    else
+                    {
+                        SetTag(target, args[0]);
+
+                        SendMessageToPC(user, "Tag set to: " + args[0] + ".");
+                    }
+                });
+
+            _builder.Create("destination")
+                .Description("Sets the DESTINATION localstring to the given value on a target.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .AvailableToAllOnTestEnvironment()
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    if (args.Length <= 0)
+                    {
+                        return "Missing arguments. Format should be: /destionation <VALUE>. Example: /destination EventEntrance";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    if (!GetIsObjectValid(target))
+                    {
+                        SendMessageToPC(user, "Target is invalid. Please target a placeable.");
+                    }
+                    else
+                    {
+                        SetEventScript(target, EventScript.Placeable_OnUsed, "teleport");
```

Comment:
This is another situation where I think we need to be more restrictive for this command. You can really hose world objects if the command is misused by the DMs.

The more I'm thinking about it, perhaps the workflow should be a single command like /CreateTeleporter DESTINATION_TAG which spawns a new placeable instead of modifying an existing one. Then you can change the graphic used after it's been spawned?

Reply:
I've restricted its use only to the Teleport Object placeable so there should be no issues with DM's accidentally messing up an area or some critical object.